src/main/kotlin/com/cognifide/gradle/aem/environment/EnvironmentOptions.kt

```diff
@@ -0,0 +1,45 @@
+package com.cognifide.gradle.aem.environment
+
+import com.cognifide.gradle.aem.common.Retry
+import com.cognifide.gradle.aem.environment.docker.DockerOptions
+import com.cognifide.gradle.aem.environment.hosts.HostsOptions
+import java.net.URI
+import kotlin.math.max
+
+class EnvironmentOptions {
+    val healthChecks = HealthChecks()
+    val hosts = HostsOptions()
+    val docker = DockerOptions()
+
+    fun healthChecks(configurer: HealthChecks.() -> Unit) {
+        healthChecks.apply(configurer)
+    }
+
+    fun hosts(config: Map<String, String>) {
+        hosts.configure(config)
+    }
+
+    fun docker(configurer: DockerOptions.() -> Unit) {
+        docker.apply(configurer)
+    }
+}
+
+class HealthChecks {
```

Comment:
please separate classes. I know that's fancy Kotlin feature but I am not using it in GAP

Reply:
OK